contracts/core/MarginRequirements.sol

```diff
@@ -0,0 +1,223 @@
+
+        initialMargin[keccak256(abi.encode(_underlying, _collateralAsset, _isPut))][_account] = _initialMargin;
+    }
+
+    /**
+     * @notice sets the maintenance margin absolute amount
+     * @dev can only be called by keeper
+     * @param _account account address
+     * @param _vaultID id of the vault
+     * @param _maintenanceMargin maintenance margin absolute amount
+     */
+    function setMaintenanceMargin(
+        address _account,
+        uint256 _vaultID,
+        uint256 _maintenanceMargin
+    ) external onlyKeeper {
+        require(_maintenanceMargin > 0, "MarginRequirements: initial margin cannot be 0");
+        require(_account != address(0), "MarginRequirements: invalid account");
+
+        maintenanceMargin[_account][_vaultID] = _maintenanceMargin;
+    }
+
+    /************************************************
+     *  MARGIN OPERATIONS
+     ***********************************************/
+
+    /**
+     * @notice clears the maintenance margin mapping
+     * @dev can only be called by OTC wrapper contract
+     * @param _account account address
+     * @param _vaultID id of the vault
+     */
+    function clearMaintenanceMargin(address _account, uint256 _vaultID) external onlyOTCWrapper {
+        delete maintenanceMargin[_account][_vaultID];
+    }
+
+    /**
+     * @notice checks if there is enough collateral to mint the desired amount of otokens
+     * @param _account account address
+     * @param _notional order notional amount
+     * @param _underlying underlying asset address
+     * @param _isPut option type the vault is selling
+     * @param _collateralAsset collateral asset address
+     * @param _collateralAmount collateral amount
+     * @return boolean value stating whether there is enough collateral to mint
+     */
+    function checkMintCollateral(
+        address _account,
+        uint256 _notional,
+        address _underlying,
+        bool _isPut,
+        uint256 _collateralAmount,
+        address _collateralAsset
+    ) external view returns (bool) {
+        uint256 collateralDecimals = uint256(ERC20Interface(_collateralAsset).decimals());
+
+        uint256 initialMarginRequired = initialMargin[keccak256(abi.encode(_underlying, _collateralAsset, _isPut))][
+            _account
+        ];
+
+        return
+            _notional.mul(initialMarginRequired).mul(10**collateralDecimals).mul(1e8) <
+            _collateralAmount.mul(oracle.getPrice(_collateralAsset)).mul(100e2).mul(1e6);
+    }
+
+    /**
+     * @notice checks if there is enough collateral to withdraw the desired amount
+     * @param _account account address
+     * @param _notional order notional amount
+     * @param _withdrawAmount desired amount to withdraw
+     * @param _otokenAddress otoken address
+     * @param _vaultID id of the vault
+     * @param _vault vault struct
+     * @return boolean value stating whether there is enough collateral to withdraw
+     */
+    function checkWithdrawCollateral(
```

Comment:
can you elaborate on decimals here as well.

Also, lets just pass in equivalent of `_vault.collateralAmounts[0]`, `_vault.collateralAssets[0]`

Reply:
Decimals: same logic as comment above - start with raw numbers and then add the same amount of decimals to both sides of the equation

I tried to do this in the beginning - the reason `vault` is used as an input variable is to avoid stack too deep error.

Even with `_getInitialMargin` and `_getMaintenanceMargin` internal functions it still doesn't allow passing the two variables individually